### PR TITLE
Update link branches to `main`

### PR DESCRIPTION
**Pull request by guidezpl**
- Update CS and googlesource.com link branches
- Update GitHub /blob/ and /tree/ links

Tested links manually and fixes a few broken or deprecated links

Added a test that validates that `master` isn't used, except for specified repos.

Part of https://github.com/flutter/flutter/issues/121564

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/buildbucket-dart/lib/src/generated/go.chromium.org/luci/buildbucket/proto/project_config.pb.dart [hunk 1]
```diff
@@ -1053,9 +1053,9 @@ class BuilderConfig extends $pb.GeneratedMessage {
   ///  Lower means more important.
   ///
   ///  The default value is configured in
-  ///  https://chrome-internal.googlesource.com/infradata/config/+/master/configs/cr-buildbucket/swarming_task_template.json
+  ///  https://chrome-internal.googlesource.com/infradata/config/+/89dede6f6a67eb06946a6009a6a88d377e957d25/configs/cr-buildbucket/swarming_task_template.json
```
[the remark this answers is not in the document]
Comment (guidezpl) [reply packages/buildbucket-dart/lib/src/generated/go.chromium.org/luci/buildbucket/proto/project_config.pb.dart:1056]: This file was deleted and this is the last commit

**Comment by guidezpl** [comment]
Hey just wondering what's holding this PR up?

**Review by parlough — APPROVED** [review]

File: CI_YAML.md [hunk 2]
```diff
@@ -80,7 +80,7 @@ targets:
 
 #
 # Devicelab example:
-# For tests that are located https://github.com/flutter/flutter/tree/master/dev/devicelab/bin/tasks:
+# For tests that are located https://github.com/flutter/flutter/tree/main/dev/devicelab/bin/tasks:
```
Comment (jmagman) [inline CI_YAML.md:83]: The default branch for flutter/flutter is master, not main.

File: CI_YAML.md [hunk 3]
```diff
@@ -419,7 +419,7 @@ release_build: "true"
   </tr>
   <tr>
     <td>shard</td>
-    <td>The shard name of the sharding target, used in the <a href="https://github.com/flutter/flutter/blob/master/dev/bots/test.dart">test.dart</a> test runner.
+    <td>The shard name of the sharding target, used in the <a href="https://github.com/flutter/flutter/blob/main/dev/bots/test.dart">test.dart</a> test runner.
```
Comment (jmagman) [inline CI_YAML.md:422]: flutter/flutter should stay master.

File: CI_YAML.md [hunk 4]
```diff
@@ -432,7 +432,7 @@ shard: web_tests
   </tr>
   <tr>
     <td>subshards</td>
-    <td>The sub shards of the sharding target, used in the <a href="https://github.com/flutter/flutter/blob/master/dev/bots/test.dart">test.dart</a> test runner.
+    <td>The sub shards of the sharding target, used in the <a href="https://github.com/flutter/flutter/blob/main/dev/bots/test.dart">test.dart</a> test runner.
```
Comment (jmagman) [inline CI_YAML.md:435]: flutter/flutter should stay master.

File: README.md [hunk 5]
```diff
@@ -54,7 +54,7 @@ made of multiple _tasks_. Tasks are _performed_ by _LUCI bots_.
 ## Getting started
 
 First, [set up a Flutter development
-environment](https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md#developing-for-flutter).
+environment](https://github.com/flutter/flutter/blob/main/CONTRIBUTING.md#developing-for-flutter).
```
Comment (jmagman) [inline README.md:57]: etc

**Review by jmagman — CHANGES_REQUESTED** [review]

File: CI_YAML.md (hunk 2 above)
Comment (guidezpl) [reply CI_YAML.md:83]: I'm anticipating the imminent (?) switch to `main` and future deprecation of `master`, according to https://github.com/flutter/flutter/issues/121564. `main` copies commits from `master` so all links are valid

**Comment by goderbauer** [comment]
(triage) @guidezpl Do you have any interest in reverting the changes related to flutter/flutter so we could land the other fixes included in this PR? The process to migrate flutter/flutter's branch to main are currently on hold.

**Comment by guidezpl** [comment]
No problem, but doesn't it make more sense to future-proof and land as-is? The `main` branch is currently an exact copy of `master`. Unless the new plan is to leave flutter/flutter on `master`

**Comment by Piinks** [comment]
(PR Triage): There is currently no work underway to migrate from master to main in flutter/flutter, and supplying main in links and resources has led folks to open PRs up against the wrong branch. That being said, @guidezpl do you have plans to return to this change?

**Pull request closed without merging**